watchtower-plugin/src/net/mod.rs

```diff
@@ -1 +1,25 @@
+use cln_plugin::messages;
+use serde::Deserialize;
 pub mod http;
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct ProxyInfo {
+    #[serde(flatten)]
+    /// The proxy data
+    inner: messages::ProxyInfo,
+    /// Whether to only send data though Tor or not
+    pub always_use: bool,
+}
+
+impl ProxyInfo {
+    pub fn new(proxy: Option<messages::ProxyInfo>, always_use: bool) -> Option<Self> {
```

Comment:
I think this would alredy do, won't it?

```rust
let wt_client = Arc::new(Mutex::new(
    WTClient::with_proxy(
        data_dir,
        tx,
        midstate.configuration().proxy.map(|proxy| {
            ProxyInfo::new(
                proxy,
                midstate.configuration().always_use_proxy.unwrap_or(false),
            )
        }),
    )
    .await,
));

[...]

impl ProxyInfo {
    pub fn new(proxy: messages::ProxyInfo, always_use: bool) -> Self {
        Self {
            inner: proxy,
            always_use,
        }
    }

    pub fn get_socks_addr(&self) -> String {
        format!("socks5h://{}:{}", self.inner.address, self.inner.port)
    }
}
```

We can add an additional check to let the user know that they've set `always-use-proxy` without a proxy, even though this already won't create a `ProxyInfo` if `cln_plugin::messages::ProxyInfo` is not set.

Reply:
> Note: we should do a sanity check on startup that there is indeed a proxy if `always_use_proxy` is set and refuse to run otherwise to avoid exposing our IP.

> We can add an additional check to let the user know that they've set always-use-proxy without a proxy, even though this already won't create a ProxyInfo if cln_plugin::messages::ProxyInfo is not set.

Nevermind about this. The sanity check is not there because we're using CLN options, and they do already do that check:


```
> cat config
dev-bitcoind-poll=0
network=regtest
#proxy=127.0.0.1:9050
always-use-proxy=true
dev-watchtower-max-retry-interval=1

> lightningd
--always-use-proxy needs --proxy
```

I may add a comment about it not to forget again.